apps/examples/testcase/le_tc/tcp_tls/tcp_tls_stress.c

```diff
@@ -0,0 +1,743 @@
+			}
+			if (time_sync) {
+				printf("ntpc_time sync done\n");
+			} else {
+				printf("ntpc_time sync fail\n");
+				clock_settime(CLOCK_REALTIME, &pre_tp);
+			}
+		}
+	} else {
+		printf("ntpc already runnig\n");
+	}
+	
+	ret = wifi_manager_disconnect_ap();
+	if (ret == WIFI_MANAGER_FAIL) {
+		printf("[WM] init failure\n");
+		return (void*)-1;
+	}
+	WM_TEST_WAIT;
+
+	ret = wifi_manager_deinit();
+	if (ret == WIFI_MANAGER_FAIL) {
+		printf("[WM] init failure\n");
+		return (void*)-1;
+	}
+	
+	if (ntpc_stop() < 0) {
+		printf("ntcp_stop() failed.\n");
+	}
+	printf("ntpc_stop() OK.\n");
+
+	return (void*)0;
+}
+#endif
+
+static int tcp_tls_signal_init(void)
+{
+	int res = pthread_mutex_init(&g_wm_mutex, NULL);
+	if (res != 0) {
+		printf(" Pthread mutex init fail(%d) (%d)\n", res, errno);
+		return -1;
+	}
+	res = pthread_cond_init(&g_wm_cond, NULL);
+	if (res != 0) {
+		printf(" Conditional mutex init fail(%d) (%d)\n", res, errno);
+		return -1;
+	}
+	return 0;
+}
+
+static void tcp_tls_signal_deinit(void)
+{
+	pthread_mutex_destroy(&g_wm_mutex);
+	pthread_cond_destroy(&g_wm_cond);
+}
+/*
+ * tcp tls connect
+ */
+TEST_SETUP(tls_connect)
+{
+	ST_START_TEST;
+	ST_EXPECT(WIFI_MANAGER_SUCCESS, wifi_manager_init(&g_wifi_callbacks));
+	wifi_manager_ap_config_s apconfig;
+	wm_get_apinfo(&apconfig);
+	ST_EXPECT(WIFI_MANAGER_SUCCESS, wifi_manager_connect_ap(&apconfig));
+	WM_TEST_WAIT;
+	ST_END_TEST;
+}
+
+TEST_TEARDOWN(tls_connect)
+{
+	ST_START_TEST;
+	ST_EXPECT(WIFI_MANAGER_SUCCESS, wifi_manager_deinit());
+	ST_END_TEST;
+}
+
+TEST_F(tls_connect)
+{
+	ST_START_TEST;
+	tls_config tls_c;
+	static mbedtls_config mbedtls_c;
+	ST_EXPECT(TCP_TLS_SUCCESS, tcp_tls_config_get(&tls_c));
+	ST_EXPECT(TCP_TLS_SUCCESS, tcp_tls_create(&mbedtls_c));
+	ST_EXPECT(TCP_TLS_SUCCESS, tcp_tls_connect(&mbedtls_c, &tls_c));
+	ST_EXPECT(TCP_TLS_SUCCESS, tcp_tls_destroy(&mbedtls_c));
+	ST_END_TEST;
+}
+
+/*
+ * tcp tls send
+ */
+TEST_SETUP(tls_send)
+{
+	ST_START_TEST;
+	ST_EXPECT(WIFI_MANAGER_SUCCESS, wifi_manager_init(&g_wifi_callbacks));
+	wifi_manager_ap_config_s apconfig;
+	wm_get_apinfo(&apconfig);
+	ST_EXPECT(WIFI_MANAGER_SUCCESS, wifi_manager_connect_ap(&apconfig));
+	WM_TEST_WAIT;
+	ST_END_TEST;
+}
+
+TEST_TEARDOWN(tls_send)
+{
+	ST_START_TEST;
+	ST_EXPECT(WIFI_MANAGER_SUCCESS, wifi_manager_deinit());
+	ST_END_TEST;
+}
+
+TEST_F(tls_send)
+{
+	ST_START_TEST;
+	tls_config tls_c;
+	unsigned int write_count = 0;
+	static mbedtls_config mbedtls_c;
+	ST_EXPECT(TCP_TLS_SUCCESS, tcp_tls_config_get(&tls_c));
+	ST_EXPECT(TCP_TLS_SUCCESS, tcp_tls_create(&mbedtls_c));
+	ST_EXPECT(TCP_TLS_SUCCESS, tcp_tls_connect(&mbedtls_c, &tls_c));
+	ST_EXPECT(TCP_TLS_SUCCESS, tcp_tls_write(&mbedtls_c, SIMPLE_GET_REQUEST, strlen((void*)SIMPLE_GET_REQUEST), &write_count));
+	ST_EXPECT(TCP_TLS_SUCCESS, tcp_tls_destroy(&mbedtls_c));
+	ST_END_TEST;
+}
+
+/*
+ * tcp tls recv
+ */
+TEST_SETUP(tls_recv)
+{
+	ST_START_TEST;
+	ST_EXPECT(WIFI_MANAGER_SUCCESS, wifi_manager_init(&g_wifi_callbacks));
```

Comment:
how about to change the position of calling wifi_manager_init() to main function of stress test. I think we don't need to call wifi_init and deinit at each test case.

Reply:
Every WiFi API call was successfully moved into the main function.